src/main/java/org/micks/champmaker/auth/UserController.java

```diff
@@ -22,20 +22,26 @@ public class UserController {
     JwtService jwtService;
 
     @PutMapping(value = "/{userId}/promote", consumes = MediaType.APPLICATION_JSON_VALUE)
-    public void promoteToAdmin(@PathVariable String userId) {
-        //TODO check if logged in user is admin
+    public void promoteToAdmin(@RequestHeader(value = "Authorization") String authorizationHeader, @PathVariable String userId) {
+        String jwtToken = JwtWebUtil.extractTokenFromHeader(authorizationHeader);
+        String loggedInUserId = jwtService.getSubject(jwtToken);
+        if (!userService.isUserAdmin(loggedInUserId)) {
+            throw new UnauthorizedException("Only admins can promote users to ADMIN.");
```

Comment:
Please add user ids to exception message (I think both userId and loggedInUserId are helphul), so they are visible in application logs. But at the same time they shouldn't be visible on client/Postman side.

Reply:
Fixed